### PR TITLE
Refactoring XslCompiledTransform

### PR DESCRIPTION
**Pull request by kronic**

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs
Comment (Wraith2) [inline src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:167]: I know this isn't your change but that trailing `!` looks weird. Do you think there's a way to annotate the Generate call with nullable attributes so it isn't needed?

Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:167]: @Wraith2 Unfortunately, there is no attribute in the annotations for this. [return: NotNullIfNull(memberName)]

Comment (Wraith2) [reply src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:167]: Hmm, sharplab seems to suggest `[return: System.Diagnostics.CodeAnalysis.NotNull]` would work even with the return type being nullable. https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMAGAsAKAGIYBXAGzImDIFMACGmK2ggrARgKbBoGcAHCAGN6ABQIBvAnRl0sAZmmyp+WWrmZMdJerordu9O2x0IdALx0YNAO50AGgAoAlADoA4oxoAnCABcaFwBuHQMZIwBORwhXABlGAHM/AAtnENUDAF9Q7IyZUPR5DQdQ/QMAbXQAdhA5dlcAESgIBJgAe14/KCFeVwBhNoQaAEEmMgBPXihegDk2vxnyMgBdULVCuuwAfjpPa18AlzXlY8MqugBlP29YBNcAUTB+P3H0sNy1XMygA

Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:167]: @Wraith2 Is it a little trickier if TypeBuilder? typeBldr is null, then notnull should be returned

Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:167]: https://github.com/dotnet/runtime/issues/59519

Comment (Wraith2) [reply src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:167]: Hmm, if it's nuanced then I'm not sure I'd have gone with the original unconditional trailing `!` because to me that's a blanket assertion that it will not be null. I'll stop poking at it and leave you to it.

**Review by Wraith2 — APPROVED** [review]
LGTM in general. As usual I'm not a fan of the unscoped using but that's for the maintainers to decide on.

File: src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:351]: why `!` hasn't been needed before but it is now?

File: src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslCompiledTransform.cs:398]: why `!` hasn't been needed before and it is now?

**Review by krwq — APPROVED** [review]
LGTM after comments are resolved

**Comment by kronic** [comment]
@krwq I`m removed unused methods. could you run tests?

**Review by krwq — APPROVED** [review]

**Comment by krwq** [comment]
none of the outerloop test failures look related so LGTM

**Pull request merged**
